lang/funcs/cidr.go

```diff
@@ -96,15 +97,11 @@ var CidrSubnetFunc = function.New(&function.Spec{
 			return cty.UnknownVal(cty.String), fmt.Errorf("invalid CIDR expression: %s", err)
 		}
 
-		// For portability with 32-bit systems where the subnet number
-		// will be a 32-bit int, we only allow extension of 32 bits in
-		// one call even if we're running on a 64-bit machine.
-		// (Of course, this is significant only for IPv6.)
-		if newbits > 32 {
-			return cty.UnknownVal(cty.String), fmt.Errorf("may not extend prefix by more than 32 bits")
+		if newbits > 64 {
```

Comment:
I don't think we need to have this check at all anymore, because the underlying CIDR library will already check to make sure that there's enough room to extend by `newbits` bits. I had originally put the `> 32` check here just to make sure this function wouldn't behave differently on 32-bit vs. 64-bit builds of Terraform, but using `*big.Int` now means that the behavior of `SubnetBig` should be the same for all architectures and so we shouldn't need to apply additional constraints in the Terraform layer here.

(When using IPv6, it would also be "valid" to start off with a very short prefix like e.g. `1::/1` and extend that by up to 127 new bits, which `SubnetBig` should accept just fine now and might be worth including a test for just to exercise the boundary cases a bit more.)